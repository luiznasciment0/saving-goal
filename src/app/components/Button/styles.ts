import styled from 'styled-components'
import { generateMedia } from 'styled-media-query'

const customMedia = generateMedia({
  small: '350px'
})

export const StyledButton = styled.button`
  font-size: 1.6rem;
  font-weight: 600;
  color: #ffffff;
  text-align: center;
  background: #1b31a8;
  padding-top: 1.8rem;
  padding-bottom: 1.8rem;
  width: 32rem;
  border-radius: 3.2rem;
  border: 0;

  ${customMedia.lessThan('small')`
    width: 26rem;
  `}
`
