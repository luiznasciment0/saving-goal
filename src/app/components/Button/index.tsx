import { StyledButton } from './styles'

interface Props {
  label: string
}

const Button = ({ label }: Props) => <StyledButton>{label}</StyledButton>

export default Button
