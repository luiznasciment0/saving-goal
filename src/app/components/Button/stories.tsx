import { Story, Meta } from '@storybook/react/types-6-0'
import Button from '.'

export default {
  title: 'Button',
  component: Button,
  argTypes: {
    label: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story = (args) => <Button label={args.label} />

Default.args = {
  label: 'Confirm'
}
