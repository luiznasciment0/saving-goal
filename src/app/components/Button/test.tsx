import { render, screen } from '@testing-library/react'

import Button from '.'

describe('<Button />', () => {
  it('should render the Button and its label', () => {
    const { container } = render(<Button>Teste</Button>)

    expect(screen.getByRole('button', { name: /Teste/i })).toHaveStyle({
      'font-size': '1.6rem',
      'font-weight': '600',
      color: '#ffffff',
      'text-align': 'center',
      background: '#1b31a8',
      'padding-top': '1.8rem',
      'padding-bottom': '1.8rem',
      width: '32rem',
      'border-radius': '3.2rem',
      border: 0
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
