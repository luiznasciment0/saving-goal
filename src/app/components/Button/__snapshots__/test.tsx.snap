// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the Button and its label 1`] = `
.c0 {
  font-size: 1.6rem;
  font-weight: 600;
  color: #ffffff;
  text-align: center;
  background: #1b31a8;
  padding-top: 1.8rem;
  padding-bottom: 1.8rem;
  width: 32rem;
  border-radius: 3.2rem;
  border: 0;
}

<button
  class="c0"
>
  Teste
</button>
`
